lib/Doctrine/DBAL/Schema/SqliteSchemaManager.php

```diff
@@ -406,10 +406,11 @@ protected function _getPortableViewDefinition($view)
     protected function _getPortableTableForeignKeysList($tableForeignKeys)
     {
         $list = [];
+        $i    = 0;
         foreach ($tableForeignKeys as $value) {
             $value = array_change_key_case($value, CASE_LOWER);
-            $name  = $value['constraint_name'];
-            if (! isset($list[$name])) {
+            $index = $value['constraint_name'] ?: $i++;
```

Comment:
Why is it necessary to mix integers and strings as array keys? Since numeric index names are technically possible (https://github.com/doctrine/dbal/pull/4247), it may create unnecessary conflicts. As far as I see, the keys are not used once the array is constructed.

Reply:
Short ternary operator is not allowed by PHPStan starting `3.0.x`. Please be more specific about what the empty `$value['constraint_name']` actually is.

Reply:
@morozov you are right, its better to use `$value['id']` which is the identifier of the foreign key on the table. This would allow unnamed foreign keys with multiple columns, which the approach with $i does not support. Good catch.

The PHPStan change to disallow `?:` but allow `??` makes no sense to me, but ok :)

Reply:
It makes perfect sense: it disallows implicit casting to boolean.

Reply:
> This would allow unnamed foreign keys with multiple columns, which the approach with $i does not support.

Indeed.